fix: Display correct regex in FieldNamesChecker message CY-160

lolgab opened the pull request.

machadoit commented:
Can you explain the semantics of the change on this PR? I was under the impression that the pattern was correct, but it was only return the wrong message, can you walk me through it?

lolgab commented:
@machadoit Yes, the pattern was mostly correct. The message, however, was always showing the `regex` even when the actual parameter used was `constantsRegex`. The refactor you see was mostly to avoid passing around a boolean called `isConstant` since now the `message` function needs it.
Instead of having a private function ( `isConstant` ) that wasn't general but was managing only a sub-case ( if you notice, before the `isConstant` function was being called only in one of the various `case`s in the first `tree.collect` ), I made it general so you can pass any node you want and it will tell you if it's a constant or not ( by looking at its parent ).
Having it general I removed all the passes around of `isConstant: Boolean = false` in the various fuctions.
I noticed that there was coded a special behaviour to not trigger the error when an escaped variable ( inside backticks, for example ``` `a-variable` ``` ) was not compliant to the regex, so I removed this code and now it's triggering for all non-matching variables, being them escaped or not.

machadoit commented:
Thank you very much for the explanation kind sir! :D
